src/ikpy/chain.py

```diff
@@ -28,17 +28,21 @@ def __init__(self, links, active_links_mask=None, name="chain", profile=''"", **
 
         # If the active_links_mask is not given, set it to True for every link
         if active_links_mask is not None:
-            if len(active_links_mask) != len(self.links):
-                raise ValueError("Your active links mask length of {} is different from the number of your links, which is {}".format(len(active_links_mask), len(self.links)))
+            if len(active_links_mask) != self.get_num_params():
+                raise ValueError("Your active links mask length of {} is different from the number of your links, which is {}".format(len(active_links_mask), self.get_num_params()))
             self.active_links_mask = np.array(active_links_mask)
             # Always set the last link to True
-            self.active_links_mask[-1] = False
+            #self.active_links_mask[-1] = False # whaat ???
```

Comment:
I guess it was because URDF last join could not be moved or something. But is can be expressed in an other way (the last join could have no parameter). So that keeping this hack is not really good to me.